dashboard/test/controllers/lessons_controller_test.rb

```diff
@@ -223,6 +226,20 @@ class LessonsControllerTest < ActionController::TestCase
     assert_equal 1, activity.position
   end
 
+  test 'update saves lesson name i18n in levelbuilder mode' do
+    File.stubs(:write).with {|filename, _| filename.end_with? 'scripts.en.yml'}.once
+    File.stubs(:write).with {|filename, _| filename.end_with? '.script_json'}.once
+    sign_in @levelbuilder
+    put :update, params: @update_params
+  end
+
+  test 'update does not save lesson name i18n without levelbuilder mode' do
+    Rails.application.config.stubs(:levelbuilder_mode).returns false
+    File.stubs(:write).raises('must not modify filesystem')
```

Comment:
Does this check for the raised exception or stub out the function raising an exception?

Reply:
the latter -- this stubs out the method by raising an exception if the method is called. I was able to repro this locally by commenting out the preceding line.

Reply:
What does this test actually test then? It doesn't seem to check for the exception or no change to the files?

Reply:
this makes sure that File.write is not called. For example, if I comment out the previous line, then levelbuilder_mode IS enabled, we DO call File.write, this line raises, and the test fails.

Reply:
the point is to ensure that in the test environment, where these methods may get called but where levelbuilder_mode is false, we should not modify the filesystem, since that could lead to merge conflicts and/or problems in other tests.

Reply:
perhaps a better way of saying it is that if I remove this conditional in the app code then the test fails: https://github.com/code-dot-org/code-dot-org/blob/1c6fc0306351d85334ffb798b7af3bfb0ceb01d3/dashboard/app/controllers/lessons_controller.rb#L42-L41

Reply:
Ahhh gotcha. Thanks for explaining!